src/msg/async/smc/SMCNetHandler.cc

```diff
@@ -0,0 +1,91 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2014 UnitedStack <haomai@unitedstack.com>
+ *
+ * Author: Haomai Wang <haomaiwang@gmail.com>
+ *
+ * Copyright (C) IBM Corp. 2025
+ *
+ * Author: Aliaksei Makarau <aliaksei.makarau@ibm.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netinet/ip.h>
+#include <netinet/tcp.h>
+#include <arpa/inet.h>
+
+#include "common/debug.h"
+#include "common/errno.h"
+#include "common/dout.h"
+#include "include/compat.h"
+#include "include/sock_compat.h"
+
+#include "SMCNetHandler.h"
+
+#define dout_subsys ceph_subsys_ms
+#undef dout_prefix
+#define dout_prefix *_dout << "SMCNetHandler "
+
+#ifdef HAVE_SMC
```

Comment:
This file isn't compiled if `HAVE_SMC` is false, so ifdefs here seem redundant.

Reply:
I created a separate branch with the fix, please have a look:
https://github.com/ibm-s390-cloud/ceph/commit/a74e716fed91320c7d8ce007812e7008163bf5d9